qiita_db/metadata_template/prep_template.py

```diff
@@ -111,7 +108,7 @@ def create(cls, md_template, raw_data, study, data_type,
 
         # Get a connection handler
         conn_handler = SQLConnectionHandler()
-        queue_name = "CREATE_PREP_TEMPLATE_%d" % raw_data.id
+        queue_name = "CREATE_PREP_TEMPLATE_%d" % study.id
```

Comment:
What would the benefit be of using a thread pool vs. just running multiple web servers? Threading adds a _lot_ of complexity, and I doubt qiita is thread-safe anyway. IPython jobs do not have shared memory, so I do not understand your concern there. Doing something random because it "seems" safer does not inherently make it safer.